src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -710,9 +722,15 @@ bool CorProfiler::IsAttached() const { return is_attached_; }
 //
 // Startup methods
 //
-HRESULT CorProfiler::RunILStartupHook(const ComPtr<IMetaDataEmit2>& metadata_emit,
-                                      const ModuleID module_id,
-                                      const mdToken function_token) {
+bool CorProfiler::ProfilerIsLoadedIntoAppDomain(AppDomainID app_domain_id) {
```

Comment:
So we are now loading per app domain, so every app domain will have this if we successfully loaded?

Reply:
Sorry, I'm not sure what you're asking. We'll use this new method as a master check to determine if the profiler is available in the requested AppDomain, so either the profiler was successfully loaded domain-neutral or inside the AppDomain. But, like before, the intent is to never do any instrumentation unless it has already been successfully loaded.

Reply:
Oh, gotcha. I missed the `managed_profiler_loaded_domain_neutral` check